Feature/devide bundle into modules

IlyaDevN opened the pull request:
Hi. Check my code, please.

IlyaDevN committed:
Module handleBurgerButtons.js was created

Functionality from intro.js was replaced to module
handleBurgerButtons.js and module was imported to script.js.

IlyaDevN committed:
Module popupRegistration.js was created

IlyaDevN committed:
Module swiperGlobalFunctionality.js was created

IlyaDevN committed:
Module swiperGlobalFunctionality was created

IlyaDevN committed:
Merge branch 'feature/devide-bundle-into-modules' of https://github.com/IlyaDevN/Hourse-club into feature/devide-bundle-into-modules

IlyaDevN committed:
Module swiperOurHorsesFunctionality.js was created

IlyaDevN committed:
Module swiperCommentsFunctionality.js was created

IlyaDevN committed:
Module sliderOurVisitors.js was created

IlyaDevN committed:
Module sliderOurVisitors was created

IlyaDevN committed:
Merge branch 'feature/devide-bundle-into-modules' of https://github.com/IlyaDevN/Hourse-club into feature/devide-bundle-into-modules

IlyaDevN committed:
Module popupRegistrationValidation.js was created

IlyaDevN committed:
Module questionFormValidation.js was created

IlyaDevN committed:
Map was put into a separate file

IlyaDevN committed:
fix: Change slide behavior with diff resolutions

The slide layout was incorrect after loading page with resolution less
than 1920 px.

IlyaDevN committed:
refactor: Change "for of" to "forEach"

Nover98 commented on index.html:
```diff
@@ -1214,14 +1214,9 @@ <h2>О нас говорят клиенты</h2>
-		<script src="js/script.js"></script>
-		<script src="js/components.js"></script>
-		<script src="js/intro.js"></script>
+		<script type="module" src="js/script.js"></script>
```
Rename main file to index.js

IlyaDevN replied:
Done

Nover98 commented on index.html:
```diff
@@ -1214,14 +1214,9 @@ <h2>О нас говорят клиенты</h2>
-		<script src="js/intro.js"></script>
+		<script type="module" src="js/script.js"></script>
+		<script src="js/mapFunctionality.js"></script>
 		<script src="js/swiper-bundle.min.js"></script>
```
you have to have only one entry point (index.js) everything else are modules as well

IlyaDevN replied:
Done

Nover98 commented on js/script.js:
```diff
@@ -0,0 +1,20 @@
+import handleBurgerButtons from "./modules/handleBurgerButtons.js";
+handleBurgerButtons();
```
read the article and redo it
https://learn.javascript.ru/modules-intro#kod-v-module-vypolnyaetsya-tolko-odin-raz-pri-importe

IlyaDevN replied:
Does it concern to all the modules?

IlyaDevN replied:
Done. It takes two commits.

Nover98 commented on js/modules/popupRegistration.js:
```diff
@@ -0,0 +1,81 @@
+		}
+	}
+	
+	function popupOpen(currentPopup, popupContent){
```
don't create function declaration in function declaration, module is already encapsulated

IlyaDevN replied:
Done

Nover98 commented on js/modules/popupRegistration.js:
```diff
@@ -0,0 +1,81 @@
+			popupClose(currentPopup, popupContent);
+		})
+		document.addEventListener("keydown", function(e){
+			if(e.code === "Escape"){
```
magic string, create constants instead

IlyaDevN replied:
Done

Nover98 replied:
Such a constant must be outside functions.
For constants kind of keybord keys, math constant values (like 3.14), css classes and so on use this style of code, please:
https://learn.javascript.ru/variables#konstanty-v-verhnem-registre

IlyaDevN replied:
Done

Nover98 commented on js/modules/swiperGlobalFunctionality.js:
```diff
@@ -0,0 +1,54 @@
+	return activeIndexes;
+}
+
+export function navigationButtonAppearance(slider){
```
Why do you need this?

IlyaDevN replied:
This function deactivate first and last navigation buttons in slider when slider takes extreme positions

Nover98 replied:
As I see, swiper marks buttons with special class, something like "swiper-button-disabled" or whatever. Not only you can set name of this class in swiper settings but also you can just use default class and style them in css. I don' understand why you add one more class.

IlyaDevN replied:
Done

Nover98 commented on js/modules/swiperGlobalFunctionality.js:
```diff
@@ -0,0 +1,54 @@
+export function setSlidesState(slider, className){
```
I would call this file something like "swiperHelpers"

IlyaDevN replied:
Done.

Nover98 commented on js/modules/swiperOurHorsesFunctionality.js:
```diff
@@ -1,4 +1,7 @@
-(function initializeOurHorsesSlider(){
+import { navigationButtonAppearance } from "./swiperGlobalFunctionality.js";
```
use one line instead of two, when you use named exports
![image](https://user-images.githubusercontent.com/37267977/214517508-59004821-af6a-44d3-b910-69adb07d6d7d.png)

IlyaDevN replied:
It's done

Nover98 commented on js/modules/swiperCommentsFunctionality.js:
```diff
@@ -59,16 +60,20 @@
 
-	const mqOver1920 = window.matchMedia("(min-width: 1920px)");
-	mqOver1920.addEventListener("change", function(){
+	const mqlOver1920 = window.matchMedia("(min-width: 1920px)");
```
Are you sure that there must be any difference in logic between, for example,  screen 1900 px and screen 1930px ?
maybe you should use one mq for min-width 1440 or something like this?

IlyaDevN replied:
Maybe you are right but it was in Figma. If you want, I'll redo it for another resolution. Waiting for an answer...

Nover98 replied:
I looked at the design, I agree, design is shit. But anyway you have min width 1920 and max width 1919. Use only min with 1920 and `!mediaQuire.matches` for screens less then 1920.

IlyaDevN replied:
Done

Nover98 commented on js/modules/questionFormValidation.js:
```diff
@@ -0,0 +1,76 @@
+	inputValidation();
+	checkboxQuestionsValidation();
+
+	function inputValidation(){
```
function's name must reflect what this function does and there must be verb in this name.
for example
```js
	function validate(){}
```
or if this function must return boolean, it can be called like this
```js
 function isValid(){}
```

IlyaDevN replied:
Done

IlyaDevN committed:
refactor: Unite 2 imports in one

IlyaDevN committed:
refactor: Rename functions with use of verbs

IlyaDevN committed:
refactor: Main js file renamed to index.js

IlyaDevN committed:
refactor: Plug Swiper as a module

IlyaDevN committed:
refactor: Plug map as a module

IlyaDevN committed:
refactor: Remove IIFE from two modules

IIFE was removed from handleBurgerButtons.js and mapFunctionality.js .
Modules were plugged correctly.

IlyaDevN committed:
refactor: Remove all IIFE from all the functions

IlyaDevN committed:
refactor: Remove magic string "Escape"

Nover98 commented on js/modules/popupRegistrationValidation.js:
```diff
@@ -0,0 +1,56 @@
+const errorTimeout = 3000;
+
+
+if(elemsToCheck.length > 0){
```
Do you really need this?
Check how forEach works if there are no elements in array or collection

IlyaDevN replied:
The forEach() method is not executed for empty elements.
Done.

Nover98 commented on js/modules/mapFunctionality.js:
```diff
@@ -0,0 +1,64 @@
+
+	});
+
+}, {once: true});
```
once: true?

How do you think, how many times will load event be triggered?

IlyaDevN replied:
Done.

Nover98 replied:
Are you sure that you need load event here?

IlyaDevN replied:
Done.

Nover98 commented on js/modules/mapFunctionality.js:
```diff
@@ -0,0 +1,64 @@
+
+	const loadPointCoords = loadPoint.getBoundingClientRect().top + window.pageYOffset -1000;
+
+	window.addEventListener("scroll", function scrollHandler(){
```
Don't use such constractions.
Initialize function separately and then add it like a handler.
Or use anonymous functions for it.

IlyaDevN replied:
Done.

Nover98 commented on js/modules/mapFunctionality.js:
```diff
@@ -0,0 +1,64 @@
+			window.removeEventListener("scroll", scrollHandler);
+		}
+
+		isScrollIgnored = true;
```
create separate helper function throttle (you can take [this one](https://learn.javascript.ru/call-apply-decorators#tormozyaschiy-throttling-dekorator) for example) and use if tor throttle here.

example: https://codepen.io/Nover89/pen/eYjKOgZ?editors=1111

move throttle to another module and then just import function here

IlyaDevN replied:
Done.

Nover98 commented on js/modules/mapFunctionality.js:
```diff
@@ -0,0 +1,64 @@
+		disableDefaultUI: true,
+	});
+
+	const marker = new google.maps.Marker({
```
Do you need these constant map and marker somewhere?

IlyaDevN replied:
const Map is used, const Marker was deleted
Done

Nover98 commented on js/modules/mapFunctionality.js:
```diff
@@ -0,0 +1,64 @@
+function initMap() {
+	const viewportWidth = window.innerWidth;
+	const screenResolutionTab = 768;
```
tabletScreeResultion

IlyaDevN replied:
Done

Nover98 commented on js/modules/mapFunctionality.js:
```diff
@@ -0,0 +1,64 @@
+window.initMap = initMap;
+
+let isScrollIgnored = false;
+const delay = 250;
```
throttledDelay

your variables must be more understandable

IlyaDevN replied:
Done

Nover98 commented on js/modules/popupRegistration.js:
```diff
@@ -0,0 +1,79 @@
+const body = document.querySelector("body");
```
document.body

IlyaDevN replied:
Done.

Nover98 replied:
you don't need `body` at all, just use document,body for it without a variable

IlyaDevN replied:
Done.

Nover98 commented on js/modules/popupRegistration.js:
```diff
@@ -0,0 +1,79 @@
+const closeTimeout = 3000;
+let unlock = true;
+
+if(popupLinks.length > 0){
```
see how loop for works when length is 0. You don't need this checking

IlyaDevN replied:
Done.

Nover98 commented on js/modules/popupRegistration.js:
```diff
@@ -0,0 +1,79 @@
+const popupLinks = document.querySelectorAll(".popup-link");
+const registerModalOverlay = document.querySelector(".register__modal_overlay");
+const registerModalContent = document.querySelector(".register__modal_content");
+const registerModalGratitude = document.querySelector(".register__modal_gratitude");
```
If all these dom elements has a common parent, you can find the parent first (especially if you need it as well) and then look for other elements inside the parent but not in the document. When you look for in the document, you iterate through all the elements in the dom every time, but when you iterate in the parent dom element you iterate much less and it works faster.

IlyaDevN replied:
Done.

Nover98 commented on js/modules/popupRegistration.js:
```diff
@@ -0,0 +1,79 @@
+		const popupLink = popupLinks[index];
+		popupLink.addEventListener("click", function(){
+
+			popupOpen(registerModalOverlay, registerModalContent);
```
Looks like these params `registerModalOverlay` and `registerModalContent` the only elems in the document.
So if you constantly use the same elems in function they are must not be params. You use params when you know that they can be different every time you call a function.

IlyaDevN replied:
Done.

Nover98 commented on js/modules/popupRegistrationValidation.js:
```diff
@@ -0,0 +1,56 @@
+	invalidElem.setAttribute("errorLock", false);
+
+	function hideTooltips(){
+		invalidElem.previousElementSibling.classList.remove("active");
```
Don't use previousElementSibling

IlyaDevN replied:
Done.

Nover98 commented on js/modules/sliderOurVisitors.js:
```diff
@@ -0,0 +1,70 @@
+
+function changeButtonView(){
+	if(sliderCurrentPosition == 0){
+		btnPrevOurVisitors.classList.add("slider_button_disabled");
```
Use disabled attribute instead of it.

IlyaDevN replied:
Done.

Nover98 commented on js/modules/sliderOurVisitors.js:
```diff
@@ -0,0 +1,70 @@
+
+	sliderThumbCurrentPosition += sliderThumbWidth;
+	if(sliderThumbCurrentPosition >= sliderThumbMaxPossiblePosition){
+		sliderThumbCurrentPosition = sliderThumbMaxPossiblePosition;
```
I think you can disable button here and don't need to check if the slide is last every time. Just add the `else` case and enable button again if it is disabled. Now you check both buttons after slide position changing, with my solution you will do it only for a one button.

IlyaDevN replied:
Done.

Nover98 commented on js/modules/sliderOurVisitors.js:
```diff
@@ -0,0 +1,70 @@
+	if(sliderCurrentPosition >= sliderMaxPossiblePosition){
+		sliderCurrentPosition = sliderMaxPossiblePosition;
+	}
+	slidesContainer.style.marginLeft = sliderCurrentPosition + "px";
```
margin is not the best property for animation. See transform instead. Read what properties in css is better to animate and what is not and why.

IlyaDevN replied:
Done.

Nover98 commented on js/modules/sliderOurVisitors.js:
```diff
@@ -0,0 +1,70 @@
+const sliderMinPossiblePosition = -(slidesQuantity - 1)*slideWidth;
+const sliderMaxPossiblePosition = 0;
+let sliderCurrentPosition = 0;
+
```
Looks like this slider doesn't transform automatically when site is resized

IlyaDevN replied:
Done.

IlyaDevN committed:
refactor: Rename functions and put them outside

IlyaDevN committed:
refactor: Get rid of siblings in validation

IlyaDevN committed:
refactor: Create global input validation

"questionFormValidation.js" was removed.
"popupRegistrationValidation.js was renamed to "inputValidation.js.
Magic srtings were removed.

IlyaDevN committed:
refactor: Rename "swiperGlobalFunctionality"

Nover98 commented on index.html:
```diff
@@ -36,12 +36,16 @@
+						</div>
+						<div class="isValid">
+							<p class="valid_tooltip figures">Вводите цифры</p>
+							<input class="validate" type="tel" placeholder="Ваш телефон" pattern="^[+\-\s()0-9]{1,20}" required>
```
Looks like you forgot $ in the end of the pattern

IlyaDevN replied:
It changes nothing

Nover98 replied:
You are just lucky.
![image](https://user-images.githubusercontent.com/37267977/218275461-7b17dd8e-7561-4629-9fbe-a08fcaa4134e.png)

If you used a pure JS regexp it wouldn't work properly.
This text would be valid: 22222222222222222222ssss

IlyaDevN replied:
Done.

Nover98 commented on css/components.css:
```diff
@@ -399,6 +399,9 @@ input.frm__checkbox_original{
 	}
 }
+.isValid{
+	align-self: center;
```
does it make sense without `display: flex;` ?

IlyaDevN replied:
His parent element is Display: flex. I checked it. Everything is ok.

Nover98 commented on index.html:
```diff
@@ -36,12 +36,16 @@
+							<p class="valid_tooltip letters">Вводите буквы</p>
+							<input class="validate" type="text" placeholder="Ваше имя" pattern="^[а-яА-ЯёЁa-zA-ZЁёЇїІіЄєҐґ']+$" required>
+						</div>
+						<div class="isValid">
```
Bad name for the class in this situation

IlyaDevN replied:
What class from all these classes did you dislike? 
valid_tooltip
letters
validate
isValid

Nover98 replied:
when I right comment, mainly it realates to specific line. This comment is placed under target line.
I meant isValid. Such a name usually is used for class that changes state and is added and removed with js. But nor for container element

IlyaDevN replied:
Done.

Nover98 commented on css/questions.css:
```diff
@@ -242,6 +278,7 @@
 	line-height: 10px;
 	margin-bottom: 25px;
 	display: inline-block;
+	align-self: center;
```
Do you check if it works properly for you when you add this property?

IlyaDevN replied:
What property did you mean? align-self works properly, but it seems to me that display: inline-block doesn't work

Nover98 replied:
Sorry, I didn't pay attention that it is align-self but not align-items, it was deep night)
But I don't understand why you need display: inline-block

IlyaDevN replied:
Deleted.
Done.

Nover98 commented on js/modules/swiperCommentsFunctionality.js:
```diff
@@ -1,74 +1,77 @@
-			nextEl: '.swiper_btn_next_2',
-			prevEl: '.swiper_btn_prev_2',
+import { navigationButtonAppearance, setSlidesState } from "./swiperHelpers.js";
+import Swiper from "./swiper-bundle.esm.browser.min.js";
```
rename like swiper-bundle.8.4.5.min.js for better reading

IlyaDevN replied:
Done.

Nover98 replied:
remove "esm.browser"

IlyaDevN replied:
I don't like this idea. Using its real name I know what file I use. If I rename it, I'd confuse in future...
swiper-bundle.min.js is an another file on swiper FTP

IlyaDevN replied:
I also didn't like the idea to rename it to swiper-bundle.8.4.5.esm.browser.min, but I've already done it. You can see the version of swiper in the first line inside swiper-bundle.esm.browser.min.js

Nover98 commented on js/modules/swiperCommentsFunctionality.js:
```diff
@@ -1,74 +1,77 @@
-			draggable: true,
+		768: {
+			slidesPerView: 2,
+			spaceBetween: 10,
```
If spaceBetween is the same on 3 from 4 resolutions, why do you write option all the time?

why do you use breakpoint for 320 if 320 is default?

IlyaDevN replied:
I deleted breakpoint, but some properties don't work if don't use them in every breakpoint. I checked it. Like this:
breakpoints: {
		768: {
			**slidesPerView: 2,**
		},
		1000: {
			**slidesPerView: 2,**
			allowTouchMove: false,
		}
I don't know why it works so
Done.

Nover98 replied:
maybe you should use 
```js
768: {
    slidesPerView: 2,
},
```

and just don't use on 1000

as far as I understand on 1000 in this case is used setting for 768, must work the same as media on css with mobile first

IlyaDevN replied:
I also thought so but it doesn't work. If I don't use breakpoint 1000 at all, it works properly but if I use it, I should duplicate "slidesPerView: 2" in this breakpoint. I don't know why it works so

Nover98 replied:
I cloned your repo and tried it, works perfect
![image](https://user-images.githubusercontent.com/37267977/216398719-ad23af46-94ce-4fb7-ad85-9c9e393d135e.png)

IlyaDevN committed:
refactor: Rewrite "inputValidation.js"

IlyaDevN committed:
refactor: Remove an unnecessary MediaQueryList

IlyaDevN committed:
refactor: Delete func. navigationButtonAppearance

Nover98 commented on css/components.css:
```diff
@@ -190,7 +190,12 @@ div .swiper-scrollbar-drag{
+.swiper-button-disabled{
+	opacity: 0.4;
+	cursor: auto;
+	--opacity-btnPrev: 0;
```
What is this for?

IlyaDevN replied:
Remade.
Done.

IlyaDevN committed:
refactor: Rename const variables to upper case

Nover98 commented on css/components.css:
```diff
@@ -193,7 +190,12 @@ div .swiper-scrollbar-drag{
 	pointer-events: none;
 }
-
+.swiper-button-disabled{
```
pointer-events: none;

IlyaDevN replied:
Deleted.
Done.

Nover98 commented on js/modules/mapFunctionality.js:
```diff
@@ -1,36 +1,37 @@
+const SCREEN_RESOLUTION = 768;
```
TABLET_SCREEN_RESOLUTION

Nover98 commented on js/modules/swiperOurHorsesFunctionality.js:
```diff
@@ -37,9 +37,9 @@ const swiper = new Swiper(".swiper-container-1", {
 
 const viewportWidth = window.innerWidth;
-const screenResolution = 1920;
+const SCREEN_RESOLUTION = 1920;
```
when I look at the name of the variable, I still can't understand what value can be in it: 100 or 768 or 1920

Nover98 commented on js/modules/inputValidation.js:
```diff
@@ -1,59 +1,32 @@
 		
-		const tooltip = elem.querySelector(".valid_tooltip");
 		const input = elem.querySelector(".validate");
 		if(input.tagName !== "INPUT") return;
```
why do you need this?

IlyaDevN replied:
Deleted.
Done.

Nover98 commented on js/modules/inputValidation.js:
```diff
@@ -1,59 +1,32 @@
 		input.addEventListener("invalid", function(event){
 			event.preventDefault();
-			if(input.getAttribute("isErrorShown") === "true") {
+			if(elem.getAttribute("data-isErrorShown") === "true") {
```
dataset

IlyaDevN replied:
Done.

Nover98 commented on js/modules/inputValidation.js:
```diff
@@ -1,59 +1,32 @@
-	invalidElem.setAttribute("isErrorShown", true);
+function showError(invalidItem){
+	invalidItem.classList.add("error");
+	invalidItem.setAttribute("data-isErrorShown", true);
```
dataset

Nover98 replied:
I can't understand, I sent you chapter with data attributes several time, did you read it? How can't you know how to use it?

IlyaDevN replied:
Done.

Nover98 commented on js/modules/inputValidation.js:
```diff
@@ -1,59 +1,32 @@
+	setTimeout(()=>{
+		invalidItem.classList.remove("error");
+		invalidItem.setAttribute("data-isErrorShown", false);
+	}, errorTimeout);
```
why do you need timeout here?

Nover98 replied:
even if you use setTimeout, the previous logic when it was outside the function, looked better. Function must do only those things its name is for. Or function could be called somthing like `hideErrorWithDelay`

IlyaDevN replied:
Put setTimeout outside the function.
Done.

Nover98 replied:
why do you need timeout?

IlyaDevN replied:
Because now tooltip is shown for 3 seconds. Later I'll redo validation logic. After correcting small mistakes, I'll do it. As for me, it works properly at the moment. I think we can resolve this comment.
Done.

Nover98 commented on js/modules/inputValidation.js:
```diff
@@ -1,59 +1,32 @@
 			}
-			showError(input, tooltip);
-			setTimeout(hideError, errorTimeout, input, tooltip);
+			showError(elem);
```
It shouldn't be showError and hideError together, looks awful

IlyaDevN replied:
I can put hideError into showError or put setTimeout outside hideError. What should I do?

IlyaDevN replied:
Put setTimeout outside the function.
Done.

Nover98 commented on css/components.css:
```diff
@@ -349,14 +346,14 @@ div .swiper-scrollbar-drag{
 	height: 1.5em;
 }
-.frm__label_checkbox{
+.frm__checkbox_original + label{
```
what does it mean "original" ?

IlyaDevN replied:
Renamed.
Done.

Nover98 commented on css/components.css:
```diff
@@ -349,14 +346,14 @@ div .swiper-scrollbar-drag{
 	}
 }
-input.frm__checkbox_original{
+.frm__checkbox_original{
```
fOrm

IlyaDevN replied:
Done.

Nover98 commented on css/components.css:
```diff
@@ -349,14 +346,14 @@ div .swiper-scrollbar-drag{
+.frm__checkbox_original{
 	position: absolute;
 	z-index: -1;
 	opacity: 0;
```
https://www.youtube.com/watch?v=E6kLaaQFctU&t=1s

see ho to xcorrectly hide an original checkbox if you want to have a custom one

IlyaDevN replied:
Done.

Nover98 replied:
When I click here, checkbox changes its state
![image](https://user-images.githubusercontent.com/37267977/217171641-d39dc4cc-d28f-4d78-8151-1ec218dc53de.png)

IlyaDevN replied:
Done.

Nover98 commented on css/components.css:
```diff
@@ -365,32 +362,32 @@ input.frm__checkbox_original{
 	user-select: none;
```
1) Looks like it doesn't work [in safari without prefix](https://caniuse.com/?search=user-select)
2) why do you need this?

IlyaDevN replied:
Deleted. Done.

IlyaDevN committed:
refactor: Optimize sliders buttons css

Nover98 commented on js/modules/popupRegistration.js:
```diff
@@ -0,0 +1,78 @@
+const scrollBarWidth = window.innerWidth - document.documentElement.clientWidth;
+const REOPEN_DELAY = 800;
+const SHOW_TIME = 3000;
+const BTN_ESC = "Escape";
```
we have such a code on our project
```js
KEYCODE = {
    ESC: "Escape"
}

if(... === KEYCODE.ESC){...}
```

IlyaDevN replied:
Done.

IlyaDevN committed:
refactor: Remake swiperOurHorses using matchMedia

IlyaDevN committed:
refactor: Delete check for empty array in forEach

IlyaDevN committed:
refactor: Delete prop. "once" in window.onload

IlyaDevN committed:
refactor: Take function outside event handler

IlyaDevN committed:
refactor: Delete const Marker in mapFunctionality

IlyaDevN committed:
refactor: Rename SCREEN_RESOLUTION

IlyaDevN committed:
refactor: Remake const body to document.body

IlyaDevN committed:
refactor: Delete unnecessary check for loop "for"

IlyaDevN committed:
refactor: Remake querySelector on more specific

IlyaDevN committed:
refactor: Use disabled attribute instead of class

IlyaDevN committed:
refactor: Delete "user-select: none"

IlyaDevN committed:
refactor: Change implementation of btn swithing

IlyaDevN committed:
refactor: Put disabled into HTML for prev. button

Nover98 commented on js/modules/sliderOurVisitors.js:
```diff
@@ -0,0 +1,90 @@
+	if(sliderCurrentPosition <= sliderMinPossiblePosition){
+		sliderCurrentPosition = sliderMinPossiblePosition;
+		this.disabled = true;
+		this.tabIndex = "-1";
```
Why do you need tabindex ?
Did you try to focus on disabled button?

IlyaDevN replied:
Deleted.
Done.

Nover98 commented on js/modules/sliderOurVisitors.js:
```diff
@@ -0,0 +1,90 @@
+		this.disabled = false;
+		this.tabIndex = "0";
+	}
+	btnPrevOurVisitors.disabled = false;
```
Why do you need this?

IlyaDevN replied:
Because by click on one button, I need to enable other button if it was disabled.

IlyaDevN replied:
Remade.
Done.

Nover98 commented on js/modules/popupRegistration.js:
```diff
@@ -0,0 +1,76 @@
+const body = document.body;
+const popupLinks = document.body.querySelectorAll(".popup-link");
```
use just document, not document.body
search through dom won't be slower but string of code will be shorter and clearer

IlyaDevN replied:
Done.

IlyaDevN committed:
refactor: Delete unnecessary params from functions

Nover98 commented on js/modules/popupRegistration.js:
```diff
@@ -0,0 +1,76 @@
+	body.style.paddingRight = scrollBarWidth + "px";
+	isOpen = false;
+
+	registerModalOverlay.addEventListener("click", function(e){
```
Please place this outside the function

IlyaDevN replied:
Done.

Nover98 commented on js/modules/popupRegistration.js:
```diff
@@ -0,0 +1,76 @@
+			popupClose(registerModalContent);
+		}
+	})
+	closeBtn.addEventListener("click", function(){
```
the same

IlyaDevN replied:
Done.

Nover98 commented on js/modules/popupRegistration.js:
```diff
@@ -0,0 +1,76 @@
+	closeBtn.addEventListener("click", function(){
+		popupClose(registerModalContent);
+	})
+	document.addEventListener("keydown", function(e){
```
you should remove this handler from event after closing popur, otherwise you continue detect each keydown event on your page for no reason

IlyaDevN replied:
Done.

Nover98 replied:
No, you didn't. You still have function declaration inside the popupOpen

IlyaDevN replied:
Done.

IlyaDevN committed:
refactor: Change margin to transform

IlyaDevN committed:
refactor: Rename swiper-bundle

IlyaDevN committed:
refactor: Delete onload event in mapFunctionality

IlyaDevN committed:
refactor: Renamed "frm" and "original"

IlyaDevN committed:
refactor: Delete "const body"

IlyaDevN committed:
refactor: Delete breakpoint 320 in swiper

IlyaDevN committed:
refactor: Delete pointer-events: none

IlyaDevN committed:
refactor: Delete "input.tagName !== "INPUT""

IlyaDevN committed:
refactor: Remake from setAttribute to dataset

IlyaDevN committed:
refactor: Put setTimout outside hideError

IlyaDevN committed:
refactor: Changed tooltip position for width: 480

IlyaDevN committed:
refactor: Remove eventListner after "Esc"

IlyaDevN committed:
refactor: Remove 2 eventListeners after work

Nover98 commented on js/modules/popupRegistration.js:
```diff
@@ -0,0 +1,76 @@
+	})
+	setTimeout(() => isOpen = true, REOPEN_DELAY);
+
+	form.addEventListener("submit", function(e){
```
Please, place this outside the function and use normal name for parameters, "event" instead of "e"

IlyaDevN replied:
Done.

Nover98 commented on js/modules/handleBurgerButtons.js:
```diff
@@ -0,0 +1,18 @@
+
+burgerBtn.addEventListener("click", function(){
+
+	menuOverlay.style.visibility = "visible";
```
use css classes for it

IlyaDevN replied:
Done.

Nover98 commented on js/modules/inputValidation.js:
```diff
@@ -0,0 +1,26 @@
+
+function showError(invalidItem){
+	invalidItem.classList.add("error");
+	invalidItem.dataset.isErrorShown = true;
```
why do you need this?

IlyaDevN replied:
Defense from durak... Disable repeated showing error,  while error is shown.  It's needed.

Nover98 replied:
![image](https://user-images.githubusercontent.com/37267977/216141157-0f9cd7bb-c185-4b78-b2b6-debb71d8d739.png)

IlyaDevN replied:
Done

Nover98 commented on js/modules/swiperCommentsFunctionality.js:
```diff
@@ -1,74 +1,65 @@
+const BTN_APPEARANCE_DELAY = 1000;
+const RESIZE_DELAY = 250;
+
+const swiperComments = new Swiper(".swiper-container-2", {
```
swiper-container-2 is absolutely uninformative name for the class

IlyaDevN replied:
Done.

Nover98 commented on js/modules/swiperHelpers.js:
```diff
@@ -0,0 +1,35 @@
+	});
+}
+
+export function getActiveIndexes(slider, realIndex){
```
do you need this function somewhere else?
if not, maybe you don't need to export this function?

IlyaDevN replied:
Done.

Nover98 commented on js/modules/swiperHelpers.js:
```diff
@@ -0,0 +1,35 @@
+		}
+	}
+	else {
+		const quantityOfActiveIndexes = slider.params.slidesPerView - 2;
```
1)2 - is a magic number, when I read the code I don't understand, what it means and why it is here.
2)looks like you not only get active slides but also have some additional logic, then rename function so that anybody can understand what it does without reading a code

IlyaDevN replied:
Done.

Nover98 commented on js/modules/swiperOurHorsesFunctionality.js:
```diff
@@ -0,0 +1,54 @@
+	if(!mqlOver1920.matches){
+		swiper.off("slideChange", slideStateHandler);
+
+		swiper.slides.forEach(slide => {
```
use separate function for it, something like `cleanSlidesState`

IlyaDevN replied:
Done.

Nover98 replied:
remove swiper.off... from the function, it must only clean slidesStates

IlyaDevN replied:
Done.

Nover98 commented on js/modules/swiperOurHorsesFunctionality.js:
```diff
@@ -0,0 +1,54 @@
+	swiper.on("slideChange", slideStateHandler);
+}
+
+mqlOver1920.addEventListener("change", function(){
```
1)place this function outside like a function declaration and put it in addEventListener
2)you don't need to check if `mqlOver1920.matches` separately, you do it in a handler. Remove checking and just launch function one time except the setting it like a handler. See the example
![image](https://user-images.githubusercontent.com/37267977/215854798-5c70ee95-5824-45c4-af8b-74e351d0f6b8.png)

IlyaDevN replied:
Done.

Nover98 commented on js/modules/swiperOurHorsesFunctionality.js:
```diff
@@ -0,0 +1,54 @@
+	speed: 1000,
+});
+
+const slideStateHandler = function(){
```
function declaration

IlyaDevN replied:
Done.

IlyaDevN committed:
refactor: Create function cleanSlidesState

IlyaDevN committed:
refactor: Delete export from getActiveIndexes

IlyaDevN committed:
refactor: Rename classes and folders

IlyaDevN committed:
refactor: Change styles to classes

IlyaDevN committed:
refactor: Put submitHandler outside the listener

IlyaDevN committed:
refactor: Remove func.declaration from popupOpen

IlyaDevN committed:
refactor: Use document instead of document.body

IlyaDevN committed:
refactor: Remake disabling buttons using "call"

IlyaDevN committed:
refactor: Delete disabling functions

IlyaDevN committed:
refactor: Rename isValid to input_container

Nover98 commented on js/modules/popupRegistration.js:
```diff
@@ -0,0 +1,87 @@
+const BTN_ESC = "Escape";
+let isOpen = true;
+
+for(let index = 0; index < popupLinks.length; index++){
```
use forEach

IlyaDevN replied:
Done.

Nover98 commented on js/modules/popupRegistration.js:
```diff
@@ -0,0 +1,87 @@
+}
+
+function emptyPlaceCloseHandler (event){
+	if(!event.target.closest(".register__modal_content")){
```
```js
if(event.target === this)
```

IlyaDevN replied:
I needed to change not only this line, for "this"  but I did it.
Done.

Nover98 replied:
Why do you need `registerModalBody` in your markup ?
Isn't `register__modal_overlay` enough?

by the way, you can use variable (`registerModalBody` in your code) instead of `this`, it is more clear for reading code

IlyaDevN replied:
Done.

Nover98 replied:
Why do you need registerModalBody in your markup ?
Isn't register__modal_overlay enough?

IlyaDevN replied:
I've already deleted registerModalBody.
Done.

Nover98 commented on js/modules/popupRegistration.js:
```diff
@@ -0,0 +1,87 @@
+
+	registerModalOverlay.classList.add("open");
+	registerModalContent.classList.add("open");
+	document.body.classList.add("stopPageScroll");
```
While you didn't find out how to stop scroll with css classes only, plese, move this logic to separate function something like `stopPageScroll()`

IlyaDevN replied:
Done.

Nover98 commented on js/modules/popupRegistration.js:
```diff
@@ -0,0 +1,87 @@
+	document.body.style.paddingRight = scrollBarWidth + "px";
+	isOpen = false;
+
+	registerModalOverlay.addEventListener("click", emptyPlaceCloseHandler)
```
place it outside the function, it is enough to add handler one time, you don't need to add and remove it every time you open and close modal window

IlyaDevN replied:
Done.

Nover98 commented on js/modules/popupRegistration.js:
```diff
@@ -0,0 +1,87 @@
+	isOpen = false;
+
+	registerModalOverlay.addEventListener("click", emptyPlaceCloseHandler)
+	closeBtn.addEventListener("click", closeBtnHandler);
```
place it outside the function, it is enough to add handler one time, you don't need to add and remove it every time you open and close modal window

IlyaDevN replied:
Done.

Nover98 commented on js/modules/popupRegistration.js:
```diff
@@ -0,0 +1,87 @@
+
+function closeBtnHandler(){
+	popupClose(registerModalContent);
+	closeBtn.removeEventListener("click", closeBtnHandler);
```
you don't need to remove the handler after closing

IlyaDevN replied:
Done.

Nover98 commented on js/modules/popupRegistration.js:
```diff
@@ -0,0 +1,87 @@
+function keyDownHandler(event){
+	if(event.code === BTN_ESC){
+		popupClose(registerModalContent);
+		document.removeEventListener("keydown", keyDownHandler);
```
you should remove the handler inside popupClose function

IlyaDevN replied:
Done.

Nover98 commented on js/modules/popupRegistration.js:
```diff
@@ -0,0 +1,87 @@
+
+	setTimeout(() => isOpen = true, REOPEN_DELAY);
+
+	form.addEventListener("submit", submitHandler);
```
why do you set handler for form submit every time you open popup?

IlyaDevN replied:
And why do we need unnecessary handler? If user want to register we'll set a handler. Am I wrong? Maybe I should to remove it after popup closing?

Nover98 replied:
You can set it once, outside the function, regardless of whether the popup is opened or not

IlyaDevN replied:
Done.

Nover98 commented on js/modules/handleBurgerButtons.js:
```diff
@@ -0,0 +1,18 @@
+burgerBtn.addEventListener("click", function(){
+
+	menuOverlay.classList.add("active");
+	menuBody.classList.add("active");
```
you don't need an `active` class for meny body, just use parent selector in css

`.menu_overlay.active .menu_body{ ... }

IlyaDevN replied:
Done.

Nover98 commented on js/modules/handleBurgerButtons.js:
```diff
@@ -0,0 +1,18 @@
+const menuOverlay = document.body.querySelector(".menu_overlay");
```
please, add closing menu by clicking overlay
![image](https://user-images.githubusercontent.com/37267977/216138360-63bfb935-c150-4373-a43c-80824b22bb11.png)

IlyaDevN replied:
Done.

Nover98 commented on js/modules/swiperCommentsFunctionality.js:
```diff
@@ -0,0 +1,165 @@
+
+function changeUnwrapBtnVisibility(){
+	
+	setTimeout(()=>{
```
setTimeout(changeBtnVisibility, BTN_APPEARANCE_DELAY);

IlyaDevN replied:
Done.

Nover98 commented on js/modules/swiperCommentsFunctionality.js:
```diff
@@ -0,0 +1,165 @@
+		}, RESIZE_DELAY);
+	});
+
+	function changeBtnVisibility(){
```
place function outside

IlyaDevN replied:
Done.

Nover98 commented on js/modules/swiperCommentsFunctionality.js:
```diff
@@ -0,0 +1,165 @@
+
+	let isThrottled = false;
+
+	window.addEventListener("resize", ()=>{
```
use media queries

IlyaDevN replied:
How can I use it here? The **change** event is fired any time the width of the document's viewport changes such that its width moves across ,for example, the 400px boundary in either direction. But I need event to fire on any resize...

Nover98 replied:
waiting for update

IlyaDevN replied:
Done.

Nover98 commented on js/modules/swiperCommentsFunctionality.js:
```diff
@@ -0,0 +1,165 @@
+	}, BTN_APPEARANCE_DELAY);
+
+	let isThrottled = false;
+	window.addEventListener("resize", ()=>{
```
use media queries

IlyaDevN committed:
refactor: Create and use new function throttle

IlyaDevN committed:
refactor: Make object KEYCODE for using Esc

IlyaDevN committed:
refactor: Move swiper.off outside cleanSlidesState

IlyaDevN committed:
refactor: Remake slideStateHandler to func. decl.

IlyaDevN committed:
refactor: Remake loop to forEach

IlyaDevN committed:
refactor: Remake emptyPlaceCloseHandler using this

IlyaDevN committed:
refactor: Replace removeEventListeners outside

IlyaDevN committed:
refactor: Put changeBtnVisibility outside

IlyaDevN committed:
refactor: Rename class from calculated to active

IlyaDevN committed:
refactor: Remake setTimeout without arrow function

IlyaDevN committed:
refactor: Remake handleBurgerButtons

IlyaDevN committed:
refactor: Remove menu_body from js in menu burger

Nover98 commented on js/modules/mapFunctionality.js:
```diff
@@ -0,0 +1,54 @@
+const SCROLL_DELAY = 250;
+const loadPoint = document.querySelector(".contacts__container");
+const mapSrc = document.getElementById("contactsBgMap").dataset.mapSrc;
+const loadPointCoords = loadPoint.getBoundingClientRect().top + window.pageYOffset -1000;
```
magic number

IlyaDevN replied:
Done.

Nover98 commented on js/modules/mapFunctionality.js:
```diff
@@ -0,0 +1,54 @@
+window.addEventListener("scroll", throttledScrollHandler);
+
+function scrollHandler(){
+	if(window.scrollY >= loadPointCoords){
```
you have scrollY and pageYOffset in one module, use something one of them

IlyaDevN replied:
Done.

Nover98 commented on js/modules/swiper-bundle.esm.browser.min.js.map:
```diff
@@ -0,0 +1 @@
+{"version":3,"file":"swiper-bundle.esm.browser.min.js","names":["isObject","obj","constructor","Object","extend","target","src","keys","forEach","key","length","ssrDocument","body","addEventListener","removeEventListener","activeElement","blur","nodeName","querySelector","querySelectorAll","getElementById","createEvent","initEvent","createElement","children","childNodes","style","setAttribute","g ...
```
???

IlyaDevN replied:
![image](https://user-images.githubusercontent.com/106959524/216597657-251d735c-1642-4758-8d8d-3df98daa2d33.png)
So I decided to load it. But I didn't know what this file need for... I'll read about it later.

Nover98 replied:
don't load this file to the repository if you don't know why you need it

IlyaDevN replied:
Done.

Nover98 commented on js/modules/swiperCommentsFunctionality.js:
```diff
@@ -0,0 +1,162 @@
+});
+swiperComments.on("slideChange", closeComments);
+
+const slideStateHandler = function(){
```
function declaration

IlyaDevN replied:
Done.

Nover98 commented on js/modules/handleBurgerButtons.js:
```diff
@@ -0,0 +1,19 @@
+	document.addEventListener("click", closeMenu);
+}
+
+function closeMenu(event){
```
there are must be 2 functions: open and close. There must not be any conditions inside. Close must only close menu, the same with open.
You can create the third function, something like this
```js
function closeByOverlay(){
   if(event.target === menuOverlay) closeMenu();
}
```

and set this function like a handler on click by overlay

IlyaDevN replied:
Done.

IlyaDevN committed:
refactor: Remake sliderOurVisitors

IlyaDevN committed:
refactor: Implement check resolution in swiper

Nover98 commented on js/modules/sliderOurVisitors.js:
```diff
@@ -0,0 +1,119 @@
+
+sliderThumb.style.width = sliderThumbWidth + "px"; 
+
+const mql320 = window.matchMedia("(min-width: 320px)");
```
Think over the idea of using percent in transform and maybe you won't need mediaqueries at all

IlyaDevN replied:
Done.

Nover98 commented on js/modules/swiperHelpers.js:
```diff
@@ -0,0 +1,58 @@
+		});
+	}
+
+	mqlOver1920.addEventListener("change", on1920changeStateHandler);
```
you use `setSlidesState` for slideChange event, it means that every time a slide is changed you set a handler.
Do you really think it is okay?

Nover98 replied:
Use such a code in more specific files like `swiperOurHorsesFunctionality.js` but not in helpers

IlyaDevN replied:
It's used in both swipers. It's general functionality. In horses and in comments.

IlyaDevN replied:
I don't agree with you again. That "change" event doesn't concern swiper. It concerns "mqlOver1920", it's MediaQueryList.

Nover98 replied:
function `setSlidesState` is called every time slideChange event fires.
in the code of this function you set the hanlder for mqlOver1920
```js
mqlOver1920.addEventListener("change", on1920changeStateHandler);
```
Do you really think it is okay?

IlyaDevN replied:
Done.

IlyaDevN committed:
Delete swiper-bundle.esm.browser.min.js.map

IlyaDevN committed:
refactor: Export mqlArray from helper.js

IlyaDevN committed:
refactor: Modify sliderOurVisitors using per cent

IlyaDevN committed:
Merge branch 'feature/devide-bundle-into-modules' of https://github.com/IlyaDevN/Hourse-club into feature/devide-bundle-into-modules

IlyaDevN committed:
refactor: Put conditions outside "closeMenu"

IlyaDevN committed:
refactor: Remove register__modal_body

IlyaDevN committed:
docs: Add .gitignore

IlyaDevN committed:
refactor: Remove magic number

Remove magic number and change variable

IlyaDevN committed:
refactor: Put emptyPlaceCloseHandler outside

Nover98 commented on index.html:
```diff
@@ -12,9 +12,9 @@
+		<div class="menu_overlay">
 			<div class="menu_body">
-				<div id="menu_close_btn" class="menu_close_btn">
+				<div class="menu_close_btn">
```
use `<button>`

IlyaDevN replied:
Done.

Nover98 replied:
I don't see the changes. Look for all such places and use buttons for them

IlyaDevN replied:
Done.

Nover98 replied:
are you kidding?
There are no changes in it!
This is your last vesrion of code
![image](https://user-images.githubusercontent.com/37267977/217881091-24de2d43-a9ee-4445-ad57-1e305e2e1bb1.png)

IlyaDevN replied:
Done.

Nover98 commented on js/modules/handleBurgerButtons.js:
```diff
@@ -0,0 +1,30 @@
+function openMenu(){
+	menuOverlay.classList.add("active");
+	document.body.classList.add("stopPageScroll");
+	document.addEventListener("click", closeMenuByOverlay);
```
why do you need `document ` for it instead of `menuOverlay` ?

IlyaDevN replied:
Done.

Nover98 replied:
place addEventListeners otside the function both on open and close, didn't you understand the principle?
I've pointed out such places already several times.

IlyaDevN replied:
Done.

Nover98 commented on js/modules/sliderOurVisitors.js:
```diff
@@ -0,0 +1,81 @@
+});
+
+function movePrevSlide(){
+	sliderCurrentShift += 100;
```
magic number as well as in other places in this document

IlyaDevN replied:
Done.

Nover98 commented on js/modules/sliderOurVisitors.js:
```diff
@@ -0,0 +1,81 @@
+
+sliderThumb.style.width = sliderThumbWidth + "px"; 
+
+btnPrevOurVisitors.addEventListener("click", function(){
```
I don't think it makes sense to set such a name for the button when all this code relates to the file with the name `sliderOurVisitors`

IlyaDevN replied:
Done.

Nover98 commented on js/modules/swiperHelpers.js:
```diff
@@ -0,0 +1,58 @@
+
+	mqlOver1920.addEventListener("change", on1920changeStateHandler);
+
+	function on1920changeStateHandler(){
```
don't use function declaration in function declaration

IlyaDevN replied:
Done.

Nover98 commented on js/modules/swiperHelpers.js:
```diff
@@ -0,0 +1,58 @@
+}
+
+function cleanSlidesState(){
+	const swiper = this;
```
you can use parameters here instead fo `this`

IlyaDevN replied:
Done.

IlyaDevN committed:
refactor: Put closeBtnHandler outside popupOpen

IlyaDevN committed:
feat: Switch off closing comments on slide change

IlyaDevN committed:
refactor: Remove "resize", add "MediaQueryList"

IlyaDevN committed:
refactor: Add LINE_CLAMP_DELAY

IlyaDevN committed:
feat: Switch off the button on transparent slides

IlyaDevN committed:
refactor: Use parameters instead fo this

IlyaDevN committed:
refactor: Put slidesStateHandlerOn1920px outside

IlyaDevN committed:
refactor: Rename buttons in "ourVisitors"

IlyaDevN committed:
refactor: Delete magic numbers

IlyaDevN committed:
refactor: Use menuOverlay instead of document

IlyaDevN committed:
refactor: Rename menu_close_btn

IlyaDevN committed:
refactor: Rename register__close_btn

IlyaDevN committed:
refactor: Rename modal_form_btn

IlyaDevN committed:
refactor: Rename menu_burger_btn

Nover98 commented on js/modules/handleBurgerButtons.js:
```diff
@@ -0,0 +1,31 @@
+	}
+}
+
+function closeMenuByButton(event){
```
why do you need it?

IlyaDevN replied:
Done.

Nover98 commented on js/modules/sliderOurVisitors.js:
```diff
@@ -0,0 +1,82 @@
+const slidesContainer = document.getElementById("ourVisitors__photo_container");
+const slidesQuantity = slidesContainer.querySelectorAll(".ourVisitors__photo").length;
+const SLIDER_MAX_POSSIBLE_POSITION = 0;
+const SHIFT = 100;
```
bad name, not enough information, maybe SLIDE_SHIFT_IN_PERCENT or something like this

IlyaDevN replied:
Done.

Nover98 commented on js/modules/swiperCommentsFunctionality.js:
```diff
@@ -0,0 +1,127 @@
+	const cardText = slide.querySelector(".card_text");
+
+	cardText.style.height = "";
+	setTimeout(() => slide.classList.remove("open"), LINE_CLAMP_DELAY);
```
use transitionend event for all such things here, try not to use setTimeout if there is better way to call a callback function

IlyaDevN replied:
Done.

Nover98 commented on js/modules/swiperCommentsFunctionality.js:
```diff
@@ -0,0 +1,127 @@
+import { mqlArray } from "./helpers.js" ;
+
+
+const slides = document.querySelectorAll(".comments_card_container");
```
you already have slides in swiper object, don't use querySelectorAll

IlyaDevN replied:
Done.

Nover98 commented on js/modules/swiperCommentsFunctionality.js:
```diff
@@ -0,0 +1,127 @@
+	});
+}
+
+function addHandlerOnResize(){
```
you call this function one time, it means that you don't need to have function for this

IlyaDevN replied:
Done.

Nover98 commented on js/modules/swiperCommentsFunctionality.js:
```diff
@@ -0,0 +1,127 @@
+changeBtnVisibilityDelayed();
+addHandlerOnResize();
+
+function addInitialClientHeigh(){
```
you call this function one time, it means that you don't need to have function for this

IlyaDevN replied:
Done.

Nover98 commented on js/modules/swiperCommentsFunctionality.js:
```diff
@@ -0,0 +1,127 @@
+	});
+}
+
+function initializeCommentsButtons(){
```
you call this function one time, it means that you don't need to have function for this

IlyaDevN replied:
Done.

IlyaDevN committed:
refactor: Rename all btn to button

IlyaDevN committed:
refactor: Add transitioned event

IlyaDevN committed:
refactor: Modify swiperCommentsFunctionality

IlyaDevN committed:
refactor: Rename SHIFT to SLIDE_SHIFT_IN_PERCENT

IlyaDevN committed:
refactor: Fix modal_form__checkbox height

IlyaDevN committed:
refactor: Replace <div> with <button>

IlyaDevN committed:
refactor: Replace eventListeners outside

IlyaDevN committed:
fix: Modify slide view

IlyaDevN committed:
refactor: Replace stopScroll to separate function

IlyaDevN committed:
refactor: Change input validation logic

IlyaDevN committed:
docs: Modify gitignore

Nover98 commented on js/modules/swiperCommentsFunctionality.js:
```diff
@@ -0,0 +1,122 @@
+setSlidesState(swiperComments);
+switchSlidesStateHandlerOn1920px(swiperComments);
+
+const slidesOriginal = swiperComments.slides;
```
you can add to `{ button, comment }` one more property called, for example `rootElem`

Nover98 commented on js/modules/swiperCommentsFunctionality.js:
```diff
@@ -0,0 +1,122 @@
+	});
+}
+
+function debounce(func, timeout = 300){
```
move it to helpers and import here

IlyaDevN replied:
Done.

Nover98 commented on js/modules/popupRegistration.js:
```diff
@@ -0,0 +1,93 @@
+	registerModalContent.classList.remove("open");
+	registerModalGratitude.classList.add("open");
+	isOpen = false;
+	setTimeout(()=>{
```
remove it and give a user an opportunity to close the thanks modal window by clicking a close button when he wants it

IlyaDevN replied:
Done.

Nover98 commented on js/modules/popupRegistration.js:
```diff
@@ -0,0 +1,93 @@
+const KEYCODE = {
+	ESC: "Escape"
+};
+let isOpen = true;
```
1. why do you need `isOpen` ?
2. even if you really need it, it looks very confusing: when modal box is closed `isOpen` is true. When you open modal box `isOpen` is getting false.
3. if you really need it, I think the var name should be `isOpened`

IlyaDevN replied:
Done.

Nover98 commented on js/modules/helpers.js:
```diff
@@ -0,0 +1,37 @@
+	return wrapper;
+}
+
+const mql320 = window.matchMedia("(min-width: 320px)");
```
use separate module for it, something like mediaQueries.js

IlyaDevN replied:
Done.

Nover98 commented on js/modules/inputValidation.js:
```diff
@@ -0,0 +1,36 @@
+const elemsToValidate = document.body.querySelectorAll(".input_container");
```
use document.querySele...
I've already written why

IlyaDevN replied:
Outdated.
Done.

Nover98 commented on js/modules/inputValidation.js:
```diff
@@ -0,0 +1,36 @@
+})
+
+function showError(event, elemContainer){
+	event.preventDefault();
```
use preventDefault outside and don't use event like a param.
I've already told you, if your function is called `showError` it must only show error, but not to do something else

Nover98 commented on index.html:
```diff
@@ -1,1227 +1,1536 @@
+				</picture>
+				<div class="questions__form_container">
+					<form class="questions__form" name="visitor_contacts" id="form" action="" enctype="text/plain"
+						tooltipTimeout="2000">
```
tooltiptimeout ?

IlyaDevN replied:
Done.

Nover98 commented on index.html:
```diff
@@ -1,1227 +1,1536 @@
+					<img class="questions__frame" src="img/questions/bg_elements/x320/frame_x320.png" alt="frame">
+				</picture>
+				<div class="questions__form_container">
+					<form class="questions__form" name="visitor_contacts" id="form" action="" enctype="text/plain"
```
do you really need id for this form?

IlyaDevN replied:
Done

Nover98 replied:
I think you can get rid of `action` and `enctype` as well.
Do you use "visitor_contacts" name in your js or somewhere else?

IlyaDevN replied:
Done.

Nover98 commented on js/modules/inputValidation.js:
```diff
@@ -0,0 +1,36 @@
+
+		hideError(elemContainer);
+
+		if(input.type === "text" && input.value === "") {
```
```js
input.addEventListener("input", ()=>{
     if(!input.value || input.checkValidity()) { 
         hideError();
     } else {
       showError(elemContainer);
     }
});
```

IlyaDevN replied:
Outdated.
Done.

Nover98 commented on js/modules/sliderOurVisitors.js:
```diff
@@ -0,0 +1,82 @@
+const slidesQuantity = slidesContainer.querySelectorAll(".ourVisitors__photo").length;
+const SLIDER_MAX_POSSIBLE_POSITION = 0;
+const SLIDE_SHIFT_IN_PERCENT = 100;
+let sliderMinPossiblePosition = -(slidesQuantity - 1) * SLIDE_SHIFT_IN_PERCENT;
```
const

IlyaDevN replied:
Done.

Nover98 replied:
where the changes?

IlyaDevN committed:
style: Modify pattern in "register__modal_overlay"

IlyaDevN committed:
style: Modify let to const in sliderOurVisitors

IlyaDevN committed:
refactor: Remake inputValidation

IlyaDevN committed:
refactor: Create separate module mediaQueries.js

IlyaDevN committed:
refactor: Put debounce into helpers.js

IlyaDevN committed:
refactor: Delete id="form"

IlyaDevN committed:
refactor: Delete tooltipTimeout="2000"

IlyaDevN committed:
refactor: Modify inputValidation using RULES

IlyaDevN committed:
refactor: Remove gratitude closing timer

Added to gratitude closing by button, ESC and click on empty space.
Removed closing timer.

IlyaDevN committed:
refactor: Add questionModalGratitude to questions

IlyaDevN committed:
refactor: Add link in adress to google map

IlyaDevN committed:
refactor: Delete second copy of checkbox_x420.svg

IlyaDevN committed:
refactor: Replace colors with variables

IlyaDevN committed:
refactor: Remove copy of menu

Nover98 commented on css/style.css:
```diff
@@ -21,3 +21,9 @@ body{
 body.stopPageScroll{
 	overflow-y: hidden;
 }
+:root{
```
move it to separete css file called "variables" and inlude like a first one before the rest

IlyaDevN replied:
Done.

Nover98 commented on js/modules/inputValidation.js:
```diff
@@ -0,0 +1,132 @@
+	})
+	isInputHandlerAdded = true;
+	
+	return checkings.every((item) => {
```
checkings.every( (item) => item);

IlyaDevN replied:
Done.

Nover98 commented on js/modules/inputValidation.js:
```diff
@@ -0,0 +1,132 @@
+	{
+		name: "name",
+		validate: function(input){
+			const regExp = /^[а-яА-ЯёЁa-zA-ZЁёЇїІіЄєҐґ']+$/
```
place regExp constans outside in the beginning of the file

```js
const REGEXP = {
     PHONE: "...",
     NAME: "..."
};
```


check the rest

IlyaDevN replied:
Done.

Nover98 commented on js/modules/inputValidation.js:
```diff
@@ -0,0 +1,132 @@
+		name: "name",
+		validate: function(input){
+			const regExp = /^[а-яА-ЯёЁa-zA-ZЁёЇїІіЄєҐґ']+$/
+			const str = input.value;
```
why do you need this temp const?
use inut.value

check the rest

IlyaDevN replied:
Done.

IlyaDevN committed:
refactor: Replace "str" with "input.value"

IlyaDevN committed:
refactor: Create const REGEXP

IlyaDevN committed:
refactor: Modify "checkings.every"

IlyaDevN committed:
refactor: Create a new file variables.css

IlyaDevN committed:
refactor: Rename handleBurgerButton to mobileMenu

IlyaDevN committed:
refactor: Remove "invalidTooltip"

IlyaDevN committed:
refactor: Modify all forms

Nover98 commented on css/components.css:
```diff
@@ -231,230 +232,230 @@ div .swiper-scrollbar-drag{
 	line-height: 10px;
 	visibility: hidden;
 	opacity: 0;
 	transition: all 0.3s;
```
always write exact properties you are going to animate, in order to avoid unpredictable situations and improve performance

(check all such places on the site)

IlyaDevN replied:
Done.

Nover98 commented on js/modules/inputValidation.js:
```diff
@@ -0,0 +1,131 @@
+let isInputHandlerAdded = false;
+
+const REGEXP = {
+	name: /^[а-яА-ЯёЁa-zA-ZЁёЇїІіЄєҐґ']+$/,
```
name and phone with capital letter

IlyaDevN replied:
Done.

Nover98 replied:
I mean capslock for all the letters, the same as 
```js
const KEYCODE = { ESC: ... }
```

IlyaDevN replied:
Done.

Nover98 commented on css/questions.css:
```diff
@@ -335,6 +301,9 @@
 	}
 }
+.form__label_checkbox p{
+	text-decoration: underline;
```
for `<p>` ?
I've always thought that it needs for `<a>`.

UPD 
I checked the markup, it must be a link of course.
And it must lead hear
![image](https://user-images.githubusercontent.com/37267977/219847197-46f18e9b-57b6-465f-9557-ee431f403fdd.png)

(check all the places)

IlyaDevN replied:
Done.

Nover98 replied:
Please, every time when I write comments, find all such places on the site.

IlyaDevN replied:
I didn't find any such places. Done.

Nover98 commented on css/intro.css:
```diff
@@ -88,34 +87,51 @@
+.menu_burger_button span:nth-of-type(3) {
 	top: calc(50% + 4px);
 }
+:root{
```
Why do you need variables for it?

IlyaDevN replied:
Because it's used in several connected properties in burger menu. Its values must be the same.

Nover98 replied:
It must be redone, remove both vars

IlyaDevN replied:
Done.

IlyaDevN committed:
refactor: Modify main page HTML + CSS

Nover98 commented on css/intro.css:
```diff
@@ -232,28 +268,6 @@
-	}
-}
 .header__contacts::after{
 	content: "Ежедневно 9:00 – 22:00";
```
it shouldn't be in css

IlyaDevN replied:
Done.

Nover98 commented on css/intro.css:
```diff
@@ -198,12 +225,29 @@
+		margin: 22px 0px 0px 83px;
+	}
+}
+.header__contacts svg{
```
set classes for tags, and try not to use tags in selectors without a special reason

check all the such places

IlyaDevN replied:
Done.

Nover98 commented on css/intro.css:
```diff
@@ -336,22 +333,24 @@
 	font-family: 'Alegreya Sans', sans-serif;
 	font-size: 13px;
 	line-height: 18px;
 	color: #646464;
```
var

IlyaDevN replied:
Done.

Nover98 commented on css/intro.css:
```diff
@@ -313,21 +309,22 @@
 	font-size: 24px;
 	line-height: 27px;
 	color: #000;
```
var, check all the places

IlyaDevN replied:
Done.

Nover98 commented on index.html:
```diff
@@ -137,7 +131,9 @@
 					<div class="header__contacts">
-						<a href="tel:+38 (000) 000-00-00" class="tel_number">+38 (000) 000-00-00 </a>
+						<a href="tel:+38 (000) 000-00-00" class="tel_number"><svg viewBox="0 0 16 16" fill="none" xmlns="http://www.w3.org/2000/svg">
+							<path fill-rule="evenodd" clip-rule="evenodd" d="M1.88541 0.510194C2.06038 0.335474 2.27048 0.199938 2.50179 0.112565C2.7331 0.0251932 2.98033 -0.0120195 3.22711 0.00339337C3.47389 0.0188063 3.71457 0.0864929 3.93321 0.201968C4.15185 0.317442 4.34346 0.478068 4.49533 0.673201L6.29027 2.97931C6.61926 3.40233 6.73526 3.95335 6.60526 4.47338L6.05828 6.66348C6.03 6.77691 6.03153 6.89573 6.0627 ...
```
I would use img instead

IlyaDevN replied:
I used png because img isn't transparent.
Done.

Nover98 replied:
use svg image

IlyaDevN replied:
Done.

IlyaDevN committed:
refactor: Add several checkings in validateField

Added an opportunity for multiple validation checkings on one filed

IlyaDevN committed:
refactor: Replace onclick with onsubmit in valid..

Change form validation using onsubmit. Remove additional checking in
RULES Change preventDefault from click on the buttons to onsubmit on
the forms.

IlyaDevN committed:
feat: Add link in questions__form to agreement

IlyaDevN committed:
refactor: Replace logo.svg with tel_logo.png

IlyaDevN committed:
refactor: Replace colors with variables

IlyaDevN committed:
refactor: Fix removed logo.svg

IlyaDevN committed:
refactor: Remove tags from CSS

IlyaDevN committed:
feat: Add more text to comments

IlyaDevN committed:
refactor: Modify transitions with exact property

IlyaDevN committed:
style: Rename name and phone to Name and Phone

IlyaDevN committed:
refactor: Replace after content with attr

IlyaDevN committed:
refactor: Change volume of text in comments

IlyaDevN committed:
refactor: Change line-clamp for all resolutions

IlyaDevN committed:
refactor: Move text-transform: uppercase

IlyaDevN committed:
refactor: Increase the area of menu__link_header

IlyaDevN committed:
refactor: Replace Menu with Меню

Nover98 commented on css/ourHorses.css:
```diff
@@ -84,11 +83,12 @@
 	box-shadow: 0px 14px 30px rgba(0, 0, 0, 0.07);
 	border-radius: 5px;
-	transition: all 1s !important;
+	transition: transform 1s,
```
Why do you need important?
try not to use important if there is another opportinities such as specificity (.slide.active)

IlyaDevN replied:
Done.

Nover98 replied:
sure?
![image](https://user-images.githubusercontent.com/37267977/222918950-bbd12828-f7f8-4937-89eb-1e920b7b2dc0.png)

IlyaDevN replied:
Done.

Nover98 commented on index.html:
```diff
@@ -1,1227 +1,1530 @@
-						<div class="ourVisitors__photo">
-							<div class="photo__item">
+					<button id="button_next_ourVisitors" class="slider_button">
+						<div class="arrow arrow_next"></div>
```
google if it is a good idea to use a `<div>` inside a `<button>`

Nover98 replied:
arrows are not in the center of the buttons
![image](https://user-images.githubusercontent.com/37267977/220743237-1c399cca-805c-4097-a61f-5a1705711c54.png)

check all the places

IlyaDevN replied:
Done.

Nover98 replied:
You still could have used one span with borders for it, you just needed to calculate transform: trranslate right

IlyaDevN replied:
Done.

Nover98 commented on index.html:
```diff
@@ -1,1227 +1,1530 @@
+				<span class="modal_close_button_span"></span>
+				<span class="modal_close_button_span"></span>
+			</button>
+			<form class="register__form" onsubmit="return false">
```
`onsubmit="return false"`      ???


check all the places

IlyaDevN replied:
Done.

Nover98 commented on css/intro.css:
```diff
@@ -466,24 +452,25 @@
 	top: 16px;
 	right: 21px;
 	cursor: pointer;
+	background-color: transparent;
```
you can reset default button's background-color in reset css file

IlyaDevN replied:
Done.

Nover98 commented on css/intro.css:
```diff
@@ -85,71 +109,91 @@
+	display: inline-block;
+	width: 100%;
 	font-size: 15px;
 	line-height: 50px;
```
use padding instead of big line-height, if somebody decided to add more text in a menu link, there will be a problem

IlyaDevN replied:
Done.

Nover98 commented on css/intro.css:
```diff
@@ -85,71 +109,91 @@
 	height: 2px;
-	background-color: #008139;
+	background-color: var(--color-green);
 	position: absolute;
```
why do you need position here?

IlyaDevN replied:
Done.

Nover98 commented on js/modules/inputValidation.js:
```diff
@@ -0,0 +1,114 @@
+}
+
+const RULES = [
+
```
please, remove useless empty strings

IlyaDevN replied:
Done.

Nover98 commented on js/modules/inputValidation.js:
```diff
@@ -0,0 +1,114 @@
+		name: "phone",
+		validate: function(input){
+
+			return REGEXP.Phone.test(input.value) && input.value.length >= 10;
```
input.value.length >= 10 ?

pattern must include it

IlyaDevN replied:
Done.

Nover98 commented on js/modules/inputValidation.js:
```diff
@@ -0,0 +1,114 @@
+
+	RULES.forEach((rule) => {
+		if(Object.hasOwn(input.dataset, rule.name)){
+			let isValid = rule.validate(input);
```
let ?

IlyaDevN replied:
Done.

Nover98 commented on js/modules/inputValidation.js:
```diff
@@ -0,0 +1,114 @@
+const forms = document.querySelectorAll("form");
```
https://learn.javascript.ru/form-elements#navigatsiya-formy-i-elementy

![image](https://user-images.githubusercontent.com/37267977/220756061-006ff308-5f30-47ed-a374-577d14334197.png)

IlyaDevN replied:
Done.

Nover98 commented on js/modules/inputValidation.js:
```diff
@@ -0,0 +1,114 @@
+}
+
+function isFormValid(form){
+	let checkings = [];
```
let ?

IlyaDevN replied:
Done.

Nover98 commented on js/modules/inputValidation.js:
```diff
@@ -0,0 +1,114 @@
+
+forms.forEach((form) => {
+	
+	form.addEventListener("submit", () => {
```
move handler to separate function `handleFormSubmit` and write this all in one line

```js
forms.forEach((form) => form.addEventListener("submit", handleFormSubmit));
```

IlyaDevN replied:
Done.

Nover98 commented on js/modules/inputValidation.js:
```diff
@@ -0,0 +1,114 @@
+	});
+}
+
+function formSubmit(form){
```
all this code must be in `handleFormSubmit` function

IlyaDevN replied:
Done.

Nover98 commented on js/modules/inputValidation.js:
```diff
@@ -0,0 +1,114 @@
+
+function formSubmit(form){
+	let submitSuccess = new Event("submitSuccess");
+	form.dispatchEvent(submitSuccess);
```
```js
form.dispatchEvent(new Event("submitSuccess"));
```

IlyaDevN replied:
Done.

Nover98 commented on js/modules/inputValidation.js:
```diff
@@ -0,0 +1,114 @@
+const forms = document.querySelectorAll("form");
+let onSubmit = false;
```
why do you need it?

check all the places

IlyaDevN replied:
Done.

Nover98 commented on js/modules/inputValidation.js:
```diff
@@ -0,0 +1,114 @@
+		onSubmit = false;
+		const isValid = validateField(input);
+
+		hideError(input);
```
```js
if(!isValid) {
	showError(element);
} else {
	hideError(element);
}
```

IlyaDevN replied:
Done.

IlyaDevN committed:
refactor: Remove scrollX for resolution 1000px

IlyaDevN committed:
refactor: Put tel icon as SVG into HTML

IlyaDevN committed:
refactor: Remake checkboxes using SVG

IlyaDevN committed:
refactor: Remove class "validate" from input

IlyaDevN committed:
refactor: Increase close modal buttons

IlyaDevN committed:
refactor: Apply text-align to reference

IlyaDevN committed:
refactor: Modify forms using padding

IlyaDevN committed:
refactor: Modify button menu_burger

IlyaDevN committed:
refactor: Autoformat CSS files

IlyaDevN committed:
refactor: Modify questions_form using flex

IlyaDevN committed:
refactor: Remove useless cursor: pointer

IlyaDevN committed:
refactor: Modify menu__link_header using padding

IlyaDevN committed:
refactor: Remove unnecessary variables in "intro"

IlyaDevN committed:
refactor: Remove properties in menu_close_button

IlyaDevN committed:
refactor: Remove xmlns from svg

IlyaDevN committed:
refactor: Remove a joke from the map

IlyaDevN committed:
refactor: Remove onsubmit return false from HTML

IlyaDevN committed:
refactor: Use document.forms

IlyaDevN committed:
refactor: Remove "let onSubmit = false;"

IlyaDevN committed:
refactor: Remove "input.value.length >= 10"

IlyaDevN committed:
refactor: Replace let with const

IlyaDevN committed:
refactor: Remove useless empty strings

IlyaDevN committed:
refactor: Make separate handleFormSubmit

IlyaDevN committed:
refactor: Put "hideError" into "else"

IlyaDevN committed:
refactor: Put formSubmit into handleFormSubmit

IlyaDevN committed:
refactor: Remove "let submitSuccess = ..."

IlyaDevN committed:
refactor: Remove !important

IlyaDevN committed:
refactor: Modify arrows in slider navigation

IlyaDevN committed:
refactor: Use svg image for tel_icon in header

IlyaDevN committed:
refactor: Rename name and phone using Caps Lock

Nover98 commented on js/modules/inputValidation.js:
```diff
@@ -0,0 +1,108 @@
+	}
+]
+
+document.onsubmit = () => {return false};
```
use 
```js
event.preventDefault();
```
in `handleFormSubmit` function

IlyaDevN replied:
Done.

Nover98 commented on js/modules/inputValidation.js:
```diff
@@ -0,0 +1,108 @@
+	return checkings.every((item) => item);
+}
+
+forms.forEach((form) => form.addEventListener("submit", handleFormSubmit));
```
put this string above the functions' initializations

IlyaDevN replied:
Done.

Nover98 commented on js/modules/inputValidation.js:
```diff
@@ -0,0 +1,108 @@
+	{
+		name: "name",
+		validate: function(input){
+
```
please, remove useless empty strings

IlyaDevN replied:
Done.

Nover98 commented on js/modules/inputValidation.js:
```diff
@@ -0,0 +1,108 @@
+document.onsubmit = () => {return false};
+
+function validateField(input){
+	let checkings = [];
```
let ?

IlyaDevN replied:
Done.

Nover98 commented on css/intro.css:
```diff
@@ -1,492 +1,543 @@
 	font-size: 15px;
-	line-height: 50px;
-	color: #FFF;
+	padding: 17.5px 0;
```
add left and right padding in order to prevent this
![image](https://user-images.githubusercontent.com/37267977/222918664-a6dcf7df-db88-4349-87f0-42960c6f17f7.png)

IlyaDevN replied:
Done.

Nover98 commented on index.html:
```diff
@@ -1,1227 +1,1579 @@
+					</nav>
+					<div class="header__contacts" data-timetable="Ежедневно 9:00 – 22:00">
+						<a class="tel_number" href="tel:+38 (000) 000-00-00">
+							<img class="tel_number_img" src="img/intro/introHeader/tel.svg" alt="tel_icon">
```
read what is alt attribute and check if you really need to join words in it

check all the places

IlyaDevN replied:
Done.

Nover98 replied:
Considering language of your site, I think there must be not English words, check all the places

IlyaDevN replied:
Done.

IlyaDevN committed:
refactor: Modify alt content

IlyaDevN committed:
refactor: Add padding to menu__link_header

IlyaDevN committed:
refactor: Remove useless empty strings

IlyaDevN committed:
refactor: Use event.preventDefault() for submit

IlyaDevN committed:
refactor: Replace let with const for checkings=[]

IlyaDevN committed:
refactor: put the string above function init...

IlyaDevN committed:
refactor: Put menu into intro

IlyaDevN committed:
refactor: Remove rebuilding due to a low speed

IlyaDevN committed:
refactor: Remove "important!"

IlyaDevN committed:
refactor: Modify arrows in sliders

IlyaDevN committed:
refactor: Remove useless attributes in forms

IlyaDevN committed:
refactor: Replace <p> with <span> inside <label>

Nover98 commented on js/modules/inputValidation.js:
```diff
@@ -0,0 +1,108 @@
+	if(isFormValid(form)) {
+
+		form.dispatchEvent(new Event("submitSuccess"));
+		// form.submit();
```
remove it

IlyaDevN replied:
Done.

Nover98 commented on js/modules/inputValidation.js:
```diff
@@ -0,0 +1,108 @@
+	const form = event.target;
+
+	if(isFormValid(form)) {
+
```
remove this empty string

IlyaDevN replied:
Done.

Nover98 commented on js/modules/inputValidation.js:
```diff
@@ -0,0 +1,108 @@
+}
+
+function addInputHandler(input){
+
```
remove this empty string

IlyaDevN replied:
Done.

Nover98 commented on js/modules/inputValidation.js:
```diff
@@ -0,0 +1,108 @@
+let isInputHandlerAdded = false;
+
+const REGEXP = {
+	NAME: /^[а-яА-ЯёЁa-zA-ZЁёЇїІіЄєҐґ']+$/,
```
Looks like the name can be with infinite length, is it okay?

IlyaDevN replied:
Done.

Nover98 commented on js/modules/inputValidation.js:
```diff
@@ -0,0 +1,108 @@
+
+const REGEXP = {
+	NAME: /^[а-яА-ЯёЁa-zA-ZЁёЇїІіЄєҐґ']+$/,
+	PHONE: /^(?! )^[+\-\s()0-9]{10,20}$/,
```
Why do you need this?
![image](https://user-images.githubusercontent.com/37267977/223636533-35d249b3-9a21-4c69-950f-0c5e8d8b7823.png)

IlyaDevN replied:
It doesn't allow space at the beginning of the string

Nover98 commented on index.html:
```diff
@@ -1,1227 +1,1567 @@
+							записаться на <b>бесплатное</b> пробное занятие</p>
+						<div class="input_container">
+							<p class="valid_tooltip write_letters">Вводите буквы</p>
+							<input class="form__input" type="text" placeholder="Ваше имя" data-name data-to-Validate>
```
you don't need to use capital letters in data attribute, anyway you will give "toValidate" in js

IlyaDevN replied:
Done.

IlyaDevN committed:
refactor: Modify arrows in slider navigation

Nover98 commented on js/modules/popupRegistration.js:
```diff
@@ -0,0 +1,87 @@
+	}, {once:true})
+}
+
+function addButtonContent(popupLink){
```
Why do you need it?

IlyaDevN replied:
Because different popups have different content. This content is taken from the attribute of the button which opens current popup. Function is universal for all popups

Nover98 replied:
if there is a different content in popups it would not be enough just to change button's name. In this case logic of the openning popup would be different.
We don't have such a case here. 

Remove it (and don't forget html attributes)

IlyaDevN replied:
Done.

IlyaDevN committed:
refactor: Modify logic of swipers

IlyaDevN committed:
refactor: Rename date-to-Validate

IlyaDevN committed:
refactor: Add restriction to the length of name

IlyaDevN committed:
refactor: Remove empty string

IlyaDevN committed:
refactor: Remove //form.submit();

IlyaDevN committed:
refactor: Rename img alt into English

IlyaDevN committed:
refactor: Change slide width in comments

IlyaDevN committed:
refactor: Put "transitioned" into HTML

IlyaDevN committed:
refactor: Repair class link_icon

IlyaDevN committed:
refactor: Change comments swiper view

IlyaDevN committed:
refactor: Change ourHorses swiper view

IlyaDevN committed:
style: Remove empty strings

IlyaDevN committed:
refactor: Modify pagination ourHorses

IlyaDevN committed:
refactor: Modify pagination comments

IlyaDevN committed:
refactor: Modify pagination ourVisitors

IlyaDevN committed:
refactor: Change margin between bullets

IlyaDevN committed:
refactor: Fix slide height for dif amount of text

IlyaDevN committed:
refactor: Remove display: inline

IlyaDevN committed:
refactor: Add shadow to slider buttons

IlyaDevN committed:
refactor: Put header__logo outside div

IlyaDevN committed:
refactor: Remove useless code from "header__nav"

IlyaDevN committed:
refactor: Modify header__logo styles using margin

IlyaDevN committed:
refactor: Remove addButtonContent()
